cirq-core/cirq/devices/thermal_noise_model.py

```diff
@@ -228,10 +230,38 @@ def __init__(
         _validate_rates(qubits, rate_dict)
         self.gate_durations_ns: dict[type, float] = gate_durations_ns
         self.rate_matrix_GHz: dict[cirq.Qid, np.ndarray] = rate_dict
+        self.heat_rate_GHz = heat_rate_GHz
+        self.cool_rate_GHz = cool_rate_GHz
+        self.dephase_rate_GHz = dephase_rate_GHz
         self.require_physical_tag: bool = require_physical_tag
         self.skip_measurements: bool = skip_measurements
         self._prepend = prepend
 
+    def _value_equality_values_(self):
+        gate_times = tuple(sorted((k, v) for k, v in self.gate_durations_ns.items()))
+        rates = tuple(
+            sorted(
+                ((q, tuple(m.reshape(-1))) for q, m in self.rate_matrix_GHz.items()),
+                key=lambda x: repr(x[0]),
+            )
+        )
+        return gate_times, rates, self.require_physical_tag, self.skip_measurements, self._prepend
+
+    def __repr__(self) -> str:
+        gate_durations = ', '.join(
+            f'{t.__module__}.{t.__name__}: {v!r}' for t, v in self.gate_durations_ns.items()
+        )
+        return (
+            "cirq.devices.ThermalNoiseModel("
+            f"qubits={set(self.rate_matrix_GHz.keys())!r}, "
+            f"gate_durations_ns={{ {gate_durations} }}, "
```

Comment:
Please add `from cirq._compat import proper_repr` to the top of the module and reuse in repr here -

```suggestion
            f"gate_durations_ns={proper_repr(self.gate_durations_ns)}, "
```

Reply:
Fixed in 0a1332d780a300be4395f2120242f4ffc36f1d0e